poll/obs/get.js

```diff
@@ -0,0 +1,189 @@
+isPoll.chooseOne = isChooseOnePoll
+isPosition.chooseOne = isChooseOnePosition
+const buildResults = require('../../results/sync/buildResults')
+const { CHOOSE_ONE } = require('../../types')
+
+// PollDoc is a mutant that initially only has the value sync false
+// Which things need to be observabel?
+//  - Positions
+//    - Errors (these are the position errors that are not valid positions that should still be displayed somehow)
+//  - ClosesAt
+//  - Results (but that's computed from positions & closing time.
+//
+// Things that are not observable and can all live in one Value which will get set just once.:
+//  - title, author, body, channel, mentions, recps, the original poll message.
+//  {
+//    sync: boolean, (initially false until the value gets set)
+//    poll: MutantValue({
+//      title: '',
+//      author: '',
+//      body: '',
+//      channel: '',
+//      mentions: '',
+//      recps: '',
+//      value: {...poll msg}
+//    }),
+//    closesAt: MutantValue('date string'),
+//    positions: MutantArray(), (sorted in causal order)
+//    results: MutantStruct(),A computed obs of positions
+//    errors
+//  }
+//  The shape of this object is different to the one returned by async.get. Ugh.
+
+module.exports = function (server) {
+  return function get (key) {
+    const myKey = server.id
+
+    const positions = MutantArray([])
+    const myPositions = MutantArray([])
+    const closingTimes = MutantArray([])
+    const sortedClosingTimes = computed(closingTimes, sort)
+    const sortedPositions = computed(positions, sort)
+    const closesAt = computed(sortedClosingTimes, (times) => {
+      const time = times.pop()
+      return time ? time.value.content.closesAt : ''
+    })
+    const myPosition = computed(myPositions, (positions) => {
+      const sortedPositions = sort(positions)
+      return sortedPositions.pop()
+    })
+
+    const poll = Value({})
+    const results = computed(sortedPositions, (positions) => {
+      const resultsErrors = buildResults({ poll: resolve(poll), positions })
+      return resultsErrors ? resultsErrors.results : {}
+    })
+
+    const errors = computed(sortedPositions, (positions) => {
+      const resultsErrors = buildResults({ poll: resolve(poll), positions })
+      return resultsErrors ? resultsErrors.errors : {}
+    })
+
+    const pollDoc = Struct({
+      sync: false,
+      positions: sortedPositions,
+      closesAt,
+      results,
+      errors,
+      poll,
+      myPosition
+    })
+
+    server.get(key, (err, value) => {
+      if (err) return err
+
+      var poll = { key, value }
+      if (!isPoll(poll)) return new Error('scuttle-poll could not get poll, key provided was not a valid poll key')
+
+      // give subscribers a chance to start listening so they don't miss updates.
+      setImmediate(function () {
+        pollDoc.poll.set(decoratePoll(poll))
+
+        const refs = PullNotify()
```

Comment:
I think this style is easy to read, but note that it converts this into push-stream as pull-notify uses pull-pushable under the hood. I don't know if this matters?

Not sure there's any other good way to split pull-streams out... 
https://github.com/scrapjs/pull-splitter looks nice but also push-based.

Reply:
Yeah good point. I mean maybe it's ok to use createBacklinksStream lots of times? Do you have a feeling whether this is heavy thing to do lots of times?